src/payment/strategies/bolt/bolt-payment-strategy.ts

```diff
@@ -1,29 +1,131 @@
 import { isNonceLike } from '../..';
 import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
-import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
 import { OrderActionCreator, OrderRequestBody } from '../../../order';
 import { OrderFinalizationNotRequiredError } from '../../../order/errors';
-import { PaymentArgumentInvalidError } from '../../errors';
+import { StoreCreditActionCreator } from '../../../store-credit';
+import { PaymentArgumentInvalidError, PaymentMethodCancelledError } from '../../errors';
+import { NonceInstrument } from '../../payment';
 import PaymentActionCreator from '../../payment-action-creator';
-import { PaymentRequestOptions } from '../../payment-request-options';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
 import PaymentStrategy from '../payment-strategy';
 
+import { BoltCheckout, BoltTransaction } from './bolt';
+import BoltScriptLoader from './bolt-script-loader';
+
 export default class BoltPaymentStrategy implements PaymentStrategy {
+    private _boltClient?: BoltCheckout;
+    private _useBoltClient: boolean = false;
+
     constructor(
         private _store: CheckoutStore,
         private _orderActionCreator: OrderActionCreator,
-        private _paymentActionCreator: PaymentActionCreator
+        private _paymentActionCreator: PaymentActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _storeCreditActionCreator: StoreCreditActionCreator,
+        private _boltScriptLoader: BoltScriptLoader
     ) { }
 
-    initialize(): Promise<InternalCheckoutSelectors> {
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { bolt, methodId } = options;
+
+        this._useBoltClient = !!(bolt && bolt.useClientScript);
```

Comment:
do you need this to be a property of the class? maybe it can just be
```js
const useBoltClient = !!(bolt && bolt.useClientScript);
```

Reply:
I used a class property to validate it easier on the execute method, if It's declared as constant it will not be usable in the execute method of the strategy